File: Binary Search.c
#include<stdio.h>
int main()
{
    int ara[]={1,4,6,8,9,11,14,15,20,25,33,83,87,97,99,100};

    int left=0;
    int right=15; //Index length,,,(n-1)

    int mid;

    int search=1;

    while(left<=right)
    {
        mid=(left+right)/2;

        if(search==ara[mid])
        {
            break;
        }
        if(search<ara[mid])
        {
            right=mid-1;
        }
        else
        {
            left=mid+1;
        }
        
    }
    if(left>right)
    {
        printf("%d is not in the array\n",search);
    }

    else
    {
        printf("%d is found in the array .It is the %d th element of the array\n",ara[mid],mid+1);
    }
}

File: Python/Binary Search.py
array=[1,4,6,8,9,11,14,15,20,25,33,83,87,97,99,100]

left=0
right=15 #Index length,,,(n-1)

search=1

while left <= right:
    mid=(left+right)//2 

    if search==array[mid]:
        break
    if search < array[mid]:
        right=mid-1
    else:
        left=mid+1

if left>right:
    print("not found")

else:
    print(f"{array[mid]} is found in the array .It is the {mid+1} th element of the array\n")



